### DISCUSSION TITLE
Putting entity linked data into relation extraction model.

### DISCUSSION THREAD

**Discussion by deanmachine88** [explosion/spaCy#11273] [Help: Coding & Implementations]
Hello! 
Working on an end to end pipeline project here and I have a question about data input to my relation extraction model.

I have been training my relation extraction model with data that has gone through a custom NER tagger and it has been working great! However, I would like to add entity linking to the pipeline to improve my results as I think the models could learn more if it knew two entities were speaking to the same underlying entity. 

I would like my input data to use the linked entity and not the entity as it is in the text. How can I parse/format my data that comes out of the entity linking trained model from prodigy to be input to the relation extraction model?

Thanks for your help!

**Comment by polm** [comment]
You should be able to do this by customizing how the relation extraction component gets candidate vectors.

One way to do that is to add a [custom hook for `Token.vector` (or `Span.vector`)](https://spacy.io/usage/processing-pipelines#custom-components-user-hooks) - you could see if a token has an `ent_kb_id`, and if so return the relevant embedding, and otherwise fall back to the default embedding.

Another thing you could do is just define a custom attribute that has the desired behaviour, and have the relation extraction component use that directly.
